crates/matrix-sdk-ui/src/timeline/mod.rs

```diff
@@ -414,6 +416,68 @@ impl Timeline {
         Ok(())
     }
 
+    /// Send a reply to the given event.
+    ///
+    /// Currently only supports `m.room.message` events whose event ID is known
+    /// and will log an error and do nothing for other events. Please check
+    /// [`EventTimelineItem::can_be_edited`] before calling this.
+    ///
+    /// # Arguments
+    ///
+    /// * `content` - The content of the reply
+    ///
+    /// * `reply_item` - The event item you want to reply to
+    ///
+    /// * `forward_thread` - Usually `Yes`, unless you explicitly want to the
+    ///   reply to show up in the main timeline even though the `reply_item` is
+    ///   part of a thread
+    ///
+    /// * `add_mentions` - Set to `Yes` if the `mentions` of `content` are
+    ///   propagated according to user intent, `No` otherwise
+    #[instrument(skip(self, new_content))]
+    pub async fn edit(
+        &self,
+        new_content: RoomMessageEventContent,
+        edit_item: &EventTimelineItem,
+    ) -> Result<(), UnsupportedEditItem> {
+        // Early returns here must be in sync with
+        // `EventTimelineItem::can_be_edited`
+        let Some(event_id) = edit_item.event_id() else {
+            return Err(UnsupportedEditItem::MISSING_EVENT_ID);
+        };
+        let TimelineItemContent::Message(original_content) = edit_item.content() else {
+            return Err(UnsupportedEditItem::NOT_ROOM_MESSAGE);
+        };
+
+        let replied_to_message =
+            original_content.in_reply_to().and_then(|details| match &details.event {
+                TimelineDetails::Ready(event) => match event.content() {
+                    TimelineItemContent::Message(msg) => Some(OriginalRoomMessageEvent {
+                        content: msg.to_content(),
+                        event_id: event_id.to_owned(),
+                        sender: event.sender.clone(),
+                        // Dummy value, not used by make_replacement
+                        origin_server_ts: MilliSecondsSinceUnixEpoch(uint!(0)),
```

Comment:
Could we still fill it, though? So that we don't have to depend on `make_replacement` changing behavior, and us forgetting to update this part of the code.

If you don't want to do this, here's an alternative: ensure in the test that the `origin_server_ts` doesn't change after editing a message.

Reply:
No, the `RepliedToEvent` doesn't contain the timestamp. This has nothing to do with the timestamp of the message being edited, it's the timestamp of the replied-to message on the original item.

Reply:
I see, in this case could we have a test that tests this for a replied-to event, and make sure the replied-to event's timestamp doesn't change?

Reply:
No, we can't test the replied-to event's timestamp for the same reason we can't copy it: It's not part of the data model 😄